dune_client/client.py

```diff
@@ -374,3 +374,26 @@ def make_public(self, query_id: int) -> None:
         """
         response_json = self._post(route=f"/query/{query_id}/unprivate")
         assert not self.get_query(int(response_json["query_id"])).meta.is_private
+
+    def upload_csv(self, table_name: str, data: str, description: str = "") -> bool:
+        """
+        https://dune.com/docs/api/api-reference/upload-data/?h=data+upload#endpoint
+        The write API allows you to upload any .csv file into Dune. The only limitations are:
+
+        - File has to be < 200 MB
+        - Column names in the table can't start with a special character or digits.
+
+        Below are the specifics of how to work with the API.
+        """
+        response_json = self._post(
+            route="/table/upload/csv",
+            params={
```

Comment:
@dsalv how do we specify the namespace the table should be in?

Reply:
right now it is not possible to customize the namespace. All publicly visible tables go as `dune_upload.{table_name}`. All tables are also duplicated as `dune.{username}.dataset_{table_name}`.

The current plan is to keep the personal namespaces and deprecate `dune_upload` soon. There is no plan to support customizable namespaces at the moment. This needs to be escalated if we think customizable namespaces are critical